File: test_frontend_backend.py
#!/usr/bin/env python3
"""
Test script to validate frontend-backend connectivity
"""
import requests
import json
import subprocess
import sys
import time

def check_backend_health():
    """Check if backend is running and healthy"""
    try:
        response = requests.get('http://localhost:5050/api/health', timeout=5)
        if response.status_code == 200:
            print("✅ Backend health check passed")
            return True
        else:
            print(f"❌ Backend health check failed: {response.status_code}")
            return False
    except requests.exceptions.RequestException as e:
        print(f"❌ Backend connection failed: {e}")
        return False

def test_dashboard_endpoint():
    """Test if dashboard endpoint responds (should require auth)"""
    try:
        # Test tenant dashboard endpoint
        response = requests.get('http://localhost:5050/api/dashboard/tenant', timeout=5)
        print(f"📊 Dashboard endpoint status: {response.status_code}")
        return response.status_code == 401  # Should require authentication
    except requests.exceptions.RequestException as e:
        print(f"📊 Dashboard endpoint status: Connection failed")
        return False

def check_frontend():
    """Check if frontend is running"""
    try:
        response = requests.get('http://localhost:3000', timeout=5)
        if response.status_code == 200:
            print("✅ Frontend is running")
            return True
        else:
            print(f"❌ Frontend check failed: {response.status_code}")
            return False
    except requests.exceptions.RequestException as e:
        print(f"❌ Frontend connection failed: {e}")
        return False

def test_login():
    """Test login with test credentials"""
    try:
        login_data = {
            "email": "tenant@example.com",
            "password": "Password123!"
        }
        response = requests.post('http://localhost:5050/api/auth/login', 
                               json=login_data, timeout=5)
        print(f"🔐 Login test status: {response.status_code}")
        if response.status_code == 200:
            print("✅ Login successful")
            return response.json()
        else:
            print(f"❌ Login failed: {response.text}")
            return None
    except requests.exceptions.RequestException as e:
        print(f"❌ Login endpoint failed: {e}")
        return None

def main():
    print("🧪 Testing Frontend-Backend Connectivity\n")
    
    # Check services
    backend_ok = check_backend_health()
    frontend_ok = check_frontend()
    auth_ok = check_auth_endpoint()
    
    if backend_ok:
        login_result = test_login()
    
    print("\n📋 Summary:")
    print(f"Backend Health: {'✅' if backend_ok else '❌'}")
    print(f"Frontend Running: {'✅' if frontend_ok else '❌'}")
    print(f"Auth Protection: {'✅' if auth_ok else '❌'}")
    
    if not backend_ok:
        print("\n🔧 Backend is not running. Start it with:")
        print("cd /Users/mjoyner/Property/backend && python wsgi.py")
    
    if not frontend_ok:
        print("\n🔧 Frontend is not running. Start it with:")
        print("cd /Users/mjoyner/Property/frontend && npm start")

if __name__ == "__main__":
    main()

File: test_frontend_backend.js
#!/usr/bin/env node
/**
 * Test script to verify frontend-backend connectivity
 * This simulates what the frontend would do when calling the backend API
 */

const axios = require('axios');

const FRONTEND_URL = 'http://localhost:3000';
const BACKEND_URL = 'http://localhost:5050/api';

async function testConnection() {
  console.log('🔍 Testing Frontend-Backend Connectivity...\n');
  
  try {
    // Test 1: Check if backend health endpoint is accessible
    console.log('1. Testing backend health endpoint...');
    const healthResponse = await axios.get(`${BACKEND_URL}/health`, {
      withCredentials: true,
      timeout: 5000
    });
    
    console.log('✅ Backend health check successful');
    console.log(`   Status: ${healthResponse.status}`);
    console.log(`   Environment: ${healthResponse.data.environment}`);
    console.log(`   Version: ${healthResponse.data.version}\n`);
    
    // Test 2: Check CORS headers
    console.log('2. Testing CORS configuration...');
    const corsHeaders = healthResponse.headers;
    console.log(`   Access-Control-Allow-Origin: ${corsHeaders['access-control-allow-origin']}`);
    console.log(`   Access-Control-Allow-Credentials: ${corsHeaders['access-control-allow-credentials']}`);
    
    if (corsHeaders['access-control-allow-origin'] === 'http://127.0.0.1:3000' || 
        corsHeaders['access-control-allow-origin'] === '*') {
      console.log('✅ CORS configured correctly for frontend\n');
    } else {
      console.log('⚠️  CORS may need adjustment for frontend origin\n');
    }
    
    // Test 3: Test with frontend origin headers
    console.log('3. Testing with frontend origin...');
    const frontendResponse = await axios.get(`${BACKEND_URL}/health`, {
      withCredentials: true,
      headers: {
        'Origin': 'http://localhost:3000',
        'User-Agent': 'Mozilla/5.0 (Test Frontend)'
      },
      timeout: 5000
    });
    
    console.log('✅ Frontend origin request successful');
    console.log(`   Response: ${JSON.stringify(frontendResponse.data, null, 2)}\n`);
    
    // Summary
    console.log('🎉 All connectivity tests passed!');
    console.log(`
📋 Summary:
   • Backend running on: ${BACKEND_URL}
   • Frontend should connect to: ${BACKEND_URL}
   • CORS properly configured
   • Health endpoint accessible
   
✨ Next steps:
   1. Open browser to http://localhost:3000
   2. Check browser console for any API errors
   3. Test login/authentication flow
    `);
    
  } catch (error) {
    console.error('❌ Connection test failed:', error.message);
    
    if (error.code === 'ECONNREFUSED') {
      console.log('\n🔧 Troubleshooting:');
      console.log('   • Make sure backend is running: PYTHONPATH=/Users/mjoyner/Property/backend python /Users/mjoyner/Property/backend/wsgi.py');
      console.log('   • Check if port 5050 is available: lsof -i :5050');
    }
    
    process.exit(1);
  }
}

// Run the test
testConnection();
